src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -5171,7 +5171,11 @@ private ScanTypeArgumentListKind ScanTypeArgumentList(NameOptions options)
                 case SyntaxKind.GreaterThanEqualsToken:  // e.g. `e is A<B> >= C`
                 case SyntaxKind.IsKeyword:               // e.g. `e is A<B> is bool`
                 case SyntaxKind.AsKeyword:               // e.g. `e is A<B> as bool`
-                    // These tokens are added to 7.5.4.2 Grammar Ambiguities in C#7
+                    // These tokens were added to 7.5.4.2 Grammar Ambiguities in C# 7.0
+                    return ScanTypeArgumentListKind.DefiniteTypeArgumentList;
+
+                case SyntaxKind.OpenBraceToken: // e.g. `e is A<B> {}`
+                    // This token was added to 7.5.4.2 Grammar Ambiguities in C# 8.0
                     return ScanTypeArgumentListKind.DefiniteTypeArgumentList;
```

Comment:
should you only do this if the `NameOptions.AfterIs` flag is set?  tagging @gafter since he knows this area best!

Reply:
@CyrusNajmabadi Wouldn't that mean that the problematic property patterns would still fail to compile in a `switch` statement and `switch` expression?